frontend/web/components/PermissionRow.tsx

```diff
@@ -0,0 +1,157 @@
+  level: string
+  projectId?: string | number
+  entityPermissions: EntityPermissions | UserPermissions
+  onSelectPermissions: (
+    key: string,
+    type: PermissionRoleType,
+    tags?: number[],
+  ) => void
+  onTogglePermission: (key: string) => void
+  limitedPermissions?: string[]
+  isTagBasedPermissions?: boolean
+  isAdmin?: boolean
+  isSaving?: boolean
+  isDebug?: boolean
+  onValueChanged: (changed: boolean) => void
+}
+
+export const PermissionRow: React.FC<PermissionRowProps> = ({
+  entityPermissions,
+  isAdmin,
+  isDebug,
+  isSaving,
+  isTagBasedPermissions,
+  level,
+  limitedPermissions = [],
+  onSelectPermissions,
+  onTogglePermission,
+  onValueChanged,
+  permission,
+  projectId,
+}) => {
+  const matchingPermission = entityPermissions.permissions.find(
+    (e) => e.permission_key === permission.key,
+  )
+
+  const hasPermission = (key: string) => {
+    if (isAdmin) return true
+    return entityPermissions.permissions.find(
+      (permission) => permission.permission_key === key,
+    )
+  }
+
+  const getPermissionRoleType = (key: string) => {
+    if (isAdmin) return PermissionRoleType.GRANTED
+    const permission = entityPermissions.permissions.find(
+      (v) => v.permission_key === key,
+    )
+
+    if (!permission) return PermissionRoleType.NONE
+
+    if (permission.tags?.length || limitedPermissions.includes(key)) {
+      return PermissionRoleType.GRANTED_FOR_TAGS
+    }
+
+    return PermissionRoleType.GRANTED
+  }
+
+  const levelUpperCase = level.toUpperCase()
+
+  const disabled =
+    level !== 'organisation' &&
+    permission.key !== `VIEW_${levelUpperCase}` &&
+    !hasPermission(`VIEW_${levelUpperCase}`)
+
+  const permissionData = entityPermissions.permissions.find(
+    (v) => v.permission_key === permission.key,
+  )
+
+  const permissionRoleType = getPermissionRoleType(permission.key)
+
+  const showDerivedPermissions =
+    isDebug && (matchingPermission as Permission)?.is_directly_granted === false
+
+  const isRowDisabled = isAdmin && !isDebug
+  const rowStyle = isRowDisabled ? { opacity: 0.5 } : {}
+
+  return (
+    <Row
+      key={permission.key}
+      style={rowStyle}
+      className='list-item list-item-sm px-3 py-2 cursor-default'
+    >
+      <Row space>
+        <Flex>
+          <strong>{Format.enumeration.get(permission.key)}</strong>
+          <div className='list-item-subtitle'>{permission.description}</div>
+          {permissionRoleType === PermissionRoleType.GRANTED_FOR_TAGS && (
+            <AddEditTags
+              projectId={`${projectId}`}
+              value={permissionData?.tags || []}
+              onChange={(v: number[]) => {
+                onValueChanged(true)
+                onSelectPermissions(
+                  permission.key,
+                  PermissionRoleType.GRANTED_FOR_TAGS,
+                  v,
+                )
+              }}
+            />
+          )}
+          {showDerivedPermissions && (
```

Comment:
It's more of a product question but I wonder whether we should show that the user has the permission granted both directly and derived permissions.

Let's say you want to remove create project from me. You would remove it directly from my user permissions without knowing that I will still have it from a role I have but wasn't shown no?

Reply:
We already support that, but we don't show a tooltip text when is a directly assigned permission + derived, but I could add a copy text for that specific case.